xen/arch/arm/coproc/plat/coproc_xxx.c

```diff
@@ -66,24 +68,52 @@ static int vcoproc_xxx_write(struct vcpu *v, mmio_info_t *info, register_t r,
         int i;
 
         /* Just inject all irqs that coproc has */
-        for ( i = 0; i < ctx.coproc->num_irqs; i++ )
-            vgic_vcpu_inject_spi(ctx.vcoproc->domain, ctx.coproc->irqs[i]);
+        for ( i = 0; i < ctx.vcoproc->mcoproc->num_irqs; i++ )
+            vgic_vcpu_inject_spi(ctx.vcoproc->domain, ctx.vcoproc->mcoproc->irqs[i].irq);
 
         if ( r & CORPOC_XXX_ENABLE )
-            vcoproc_scheduler_vcoproc_wake(ctx.coproc->sched, ctx.vcoproc);
+            vcoproc_scheduler_vcoproc_wake(ctx.vcoproc->mcoproc->sched, ctx.vcoproc);
         else
-            vcoproc_scheduler_vcoproc_sleep(ctx.coproc->sched, ctx.vcoproc);
+            vcoproc_scheduler_vcoproc_sleep(ctx.vcoproc->mcoproc->sched, ctx.vcoproc);
```

Comment:
The constructions like these ctx.vcoproc->mcoproc->sched, ctx.vcoproc->mcoproc->irqs[i].irq, etc are hard to understand. Wouldn't be more easy to create a local pointer to ctx.vcoproc->mcoproc and use them?